### PR TITLE
Check table of content of markdown files with CI

### PR DESCRIPTION
**Pull request by MrChico**
The out of sync table of content problems started bothering me enough to actually do something about it. Addresses part of https://github.com/ethereum/eth2.0-specs/issues/1369
~Not sure if `master` is the appropriate target for this PR, I'll rebase if needed.~
Target updated to `v09x`

The diff is large, but the actual changes are small:
- `check_toc` makefile rule, fails if TOCs are out of date
- add `check_toc` to CI ~(please check this)~ (seems to work as intended now)
- Page titles and `table of contents` headers are removed from TOC
- Consistent header level for all files (which adds plenty of headers to `p2p-interface.md`)
- Fix malformed TOCs (`compute_slots_since_epoch_start`, `should_update_justified_checkpoint` in `Fork_choice` added, `BLS signature verification` removed from `bls_signature.md`).

Besides CI usage of `check_toc`, it can also be run _twice_ locally to update TOC of all relevant markdown files (requires `npm install -g doctoc` before run).

### CODE REVIEW

**Comment by djrtwo** [comment]
`v09x`  would be appropriate! it will be released into master with the next release and then I'll backport it to `dev`

File: specs/networking/p2p-interface.md [hunk 1]
```diff
@@ -18,23 +18,88 @@ It consists of four main sections:
...
+    - [Integration into libp2p stacks](#integration-into-libp2p-stacks)
+    - [ENR structure](#enr-structure)
+      - [Interop](#interop-5)
+      - [Mainnet](#mainnet-5)
+    - [Topic advertisement](#topic-advertisement)
+      - [Interop](#interop-6)
+      - [Mainnet](#mainnet-6)
 - [Design decision rationale](#design-decision-rationale)
   - [Transport](#transport-1)
+    - [Why are we defining specific transports?](#why-are-we-defining-specific-transports)
```
Comment (djrtwo) [inline specs/networking/p2p-interface.md]: Thoughts on the depth/verbosity here?
Can we tune `doctoc` depth on a per-TOC basis ?

Comment (djrtwo) [reply specs/networking/p2p-interface.md]: Hm, maybe having all of the questions explicitly in TOC is fine..

Comment (MrChico) [reply specs/networking/p2p-interface.md]: We could make a custom flag for this file, but yeah it's also nice to have verbose TOCs for large files

Comment (djrtwo) [reply specs/networking/p2p-interface.md]: verbose is fine 👍

File: specs/bls_signature.md [hunk 2]
```diff
@@ -6,23 +6,25 @@
 
 ## Table of contents
 <!-- TOC -->
```
Comment (arnetheduck) [inline specs/bls_signature.md]: is this `TOC` comment still needed?

Comment (MrChico) [reply specs/bls_signature.md]: not for `doctoc`, but it looks like some text editors adds such a comment:https://github.com/nok/markdown-toc https://github.com/AlanWalk/Markdown-TOC

**Pull request merged**